client/packages/invoices/src/InboundShipment/ListView/AppBarButtons.tsx

```diff
@@ -46,13 +61,47 @@ export const AppBarButtons: FC<{
     success(t('success'))();
   };
 
+  const createInvoice = async (nameId: string) => {
+    const invoiceNumber = await onCreate({
+      id: FnUtils.generateUUID(),
+      otherPartyId: nameId,
+    });
+
+    navigate(
+      RouteBuilder.create(AppRoute.Replenishment)
+        .addPart(AppRoute.InboundShipment)
+        .addPart(String(invoiceNumber))
+        .build()
+    );
+  };
+  useEffect(() => {
+    if (name && (data?.totalCount === 0 || !manuallyLinkInternalOrder)) {
+      createInvoice(name.id);
+    }
+  }, [name, data]);
```

Comment:
Tried to think of a way of doing this without useEffect... but.. yeah can't call in the supplier's modal onChange since it will never stay on the 'list view' page